source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,208 @@
+
+      {
+          "name": "insertOne",
+          "object": { "database": "db", "collection": "other" }
+          "documents": [ { "_id": 1 }]
+      }
+
+  inserts in the ``db.other`` collection.
+
+Each YAML file has the following keys:
+
+- ``minServerVersion``: Optional, the minimum server version (inclusive)
+  required to successfully run the test. If this field is not present, it should
+  be assumed that there is no lower bound on the required server version.
+
+- ``database_name``: The default database name to use in operations. This may be overridden in a per operation ``object``.
+
+- ``collection_name``: The default collection to use in operations. This may be overridden in a per operation ``object``.
+
+- ``encryption_setup``: Setup that should occur for each test.
+
+  - ``collections``: Describes collections that should be created and prepopulated.
+
+    - ``database``: The database for the collection to create.
+
+    - ``collection``: The name of the collection to create.
+
+    - ``data``: A list of documents, or a file path, to insert. If this is an encrypted
+      collection (specified by the jsonSchema) the data should be inserted *without*
+      automatic encryption (i.e. create a client without configuring encryption to
+      insert this data).
+
+    - ``remoteSchema``: A document or file path for the schema that should be configured remotely on the server.
+
+    - ``external``: If true, create this collection on the external key vault.
+
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+
+  - ``description``: The name of the test.
+
+  - ``skipReason``: Optional, string describing why this test should be
+    skipped.
+
+  - ``clientOptions``: Optional, parameters to pass to MongoClient().
+
+    - ``clientSideEncryptionOptions``
+
+      - ``kmsProvider``: Fixed as "aws-kms".
+
+      - ``kmsProviderOptions``: Optional.
+
+        - ``uri``: Optional. If set to "localhost:12345", use a local mock KMS service. (TODO: mock KMS not available yet.) 
+
+      - ``extraOptions``:
+
+      - ``schemas``: An optional array of schemas for local configuration.
+
+        - ``database``: The database name for the target collection.
+
+        - ``collection``: The name of the target collection.
+
+        - ``schema``: The schema document.
+
+      - ``useRemoteSchemas``: An optional boolean. If true, schemas are retrieved from the MongoDB server using listCollections.
+
+  - ``kmsPreCmd``: Optional, a command to send to the mock KMS server
+    before the test starts.
+
+  - ``kmsPostCmd``: Optional, a command to send to the mock KMS server
+    after the test ends.
+
+  - ``operations``: Array of documents, each describing an operation to be
+    executed. Each document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: Optional. If present, indicates the target collection.
```

Comment:
I think this should note specific strings to expect, such as "collection1" and "collection2", as is done in https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#test-format re: multiple session values. That would also signal to spec runners that they need not expect strings like "database" and "collection" here.